### ISSUE TITLE
Navigation on mobile usability bug - side menu is too deep and too far on mobile browser

### ISSUE THREAD

**Issue by marisveide** [Shopify/shopify-app-bridge#153]
## Describe the bug

It is basically a usability bug - inconsistency between the Shopify mobile app and the app in mobile browser.
Basically, when migrating all the navigation to the side-menu `<NavigationMenu>` component - then on the mobile web browser this side menu is too far, too hard to find for normal use.
Especially, if a user has several apps pinned, so she has to go to the _hamburger_ menu of the Admin, then scroll down the side menu, and only then she will find the app Navigation menu item.

### To Reproduce

Steps to reproduce the behaviour:

1. Create a new Shopify bootstrap application, following your own tutorial: https://github.com/Shopify/shopify-app-template-ruby/blob/main/README.md
2. Open the app on the mobile web browser.
3. Try to use the NavigationMenu while in that app.

![1 - NavigationMenu on mobile browser](https://user-images.githubusercontent.com/17713131/194643306-25669cec-929e-40e2-8b0b-64e7b03c945c.jpeg)

----

### Expected behaviour

Whereas, on the Shopify mobile app - the `<NavigationMenu>` component generates a very convenient menu on the top of the same view, without having to leave the app view.

![2 - NavigationMenu on Shopify mobile app](https://user-images.githubusercontent.com/17713131/194643325-585bc71a-094f-47db-a51b-bc4bc63ea08e.jpeg)

----

### Packages and versions

- `@shopify/app-bridge` @ `^3.1.0`
- `@shopify/app-bridge-react` @ `^3.1.0`
- `@shopify/app-bridge-utils` @ `^3.1.0`
- `@shopify/polaris` @ `^9.11.0`
- `react-router-dom` @ `^6.3.0`

## Platform

- OS: iOS
- OS Version: 15.6.1
- App: Chrome and Safari

### Additional context

We are following those requirements that say that the NavigationMenu and the page action buttons must not repeat each other:
https://shopify.dev/apps/design-guidelines/navigation

<img width="835" alt="Screenshot 2022-10-07 at 22 53 21" src="https://user-images.githubusercontent.com/17713131/194642814-8010e0ee-446a-4227-abbc-65e60b2c038b.png">

So if we ONLY have the NavigationMenu for navigation, then the mobile user experience will be dramatically decreased.
Having navigation as page actions on mobile looks a lot more convenient, with the current AppBridge behaviour.
Good solution would be if the AppBridge on mobile browser would render similar experience for NavigationMenu as it does inside the Shopify mobile app.

**Comment by kaarelss** [comment]
Agree, it will be really frustrating for merchants to find the NavigationMenu in mobile.

**Comment by marisveide** [comment]
@itissible/matrixify
